sdk/eventhub/event-hubs/src/eventHubSender.ts

```diff
@@ -217,68 +217,54 @@ export class EventHubSender extends LinkEntity {
       abortSignal?: AbortSignalLike;
     } = {}
   ): Promise<number> {
-    const abortSignal = options.abortSignal;
-    const retryOptions = options.retryOptions || {};
     if (this.isOpen()) {
       return this._sender!.maxMessageSize;
     }
-    return new Promise<number>(async (resolve, reject) => {
-      const rejectOnAbort = (): void => {
-        const desc: string = `[${this._context.connectionId}] The create batch operation has been cancelled by the user.`;
-        // Cancellation is user-intented, so treat as info instead of warning.
-        logger.info(desc);
-        const error = new AbortError(`The create batch operation has been cancelled by the user.`);
-        reject(error);
-      };
-
-      const onAbort = (): void => {
-        if (abortSignal) {
-          abortSignal.removeEventListener("abort", onAbort);
-        }
-        rejectOnAbort();
-      };
-
-      if (abortSignal) {
-        // the aborter may have been triggered between request attempts
-        // so check if it was triggered and reject if needed.
-        if (abortSignal.aborted) {
-          return rejectOnAbort();
-        }
-        abortSignal.addEventListener("abort", onAbort);
-      }
+
+    const retryOptions = options.retryOptions || {};
+    const timeoutInMs = getRetryAttemptTimeoutInMs(retryOptions);
+    retryOptions.timeoutInMs = timeoutInMs;
+    const senderOptions = this._createSenderOptions(timeoutInMs);
+
+    const createLinkPromise = async (): Promise<void> => {
       try {
-        logger.verbose(
-          "Acquiring lock %s for initializing the session, sender and " +
-            "possibly the connection.",
-          this.senderLock
-        );
-        const senderOptions = this._createSenderOptions(Constants.defaultOperationTimeoutInMs);
-        await defaultLock.acquire(this.senderLock, () => {
-          const config: RetryConfig<void> = {
-            operation: () => this._init(senderOptions),
-            connectionId: this._context.connectionId,
-            operationType: RetryOperationType.senderLink,
-            abortSignal: abortSignal,
-            retryOptions: retryOptions
-          };
-
-          return retry<void>(config);
+        await waitForTimeoutOrAbortOrResolve({
+          actionFn: () => {
+            return defaultLock.acquire(this.senderLock, () => {
+              return this._init(senderOptions);
+            });
```

Comment:
We almost don't need the waitForTimeoutOrAbortOrResolve helper. I'll be adding abortSignal support to `this._init`, and `acquire` supports passing in a timeout (though we'd have to massage the timeout error and make `_init` honor it as well.)

Reply:
Is this comment just for this part of the code or other places where we are using waitForTimeoutOrAbortOrResolve ?

Reply:
But `acquire` does not take abortSignal, so we might still need this?

Reply:
Ah yeah, we'd need our own version of acquire that supported cancellation. Maybe something to look into for another day. It'd be nice to be able to remove the callback function from the lock's internal queue if it was cancelled so it never even gets invoked then.